packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -1670,12 +1670,19 @@ interface ScalableAttributePair {
  */
 class SourceTableAttachedPolicy extends CoreConstruct implements iam.IGrantable {
   public readonly grantPrincipal: iam.IPrincipal;
-  public readonly policy: iam.IPolicy;
+  public readonly policy: iam.IManagedPolicy;
 
   public constructor(sourceTable: Table, role: iam.IRole) {
-    super(sourceTable, `SourceTableAttachedPolicy-${Names.nodeUniqueId(role.node)}`);
-
-    const policy = new iam.Policy(this, 'Resource', { roles: [role] });
+    super(sourceTable, `SourceTableAttachedManagedPolicy-${Names.nodeUniqueId(role.node)}`);
+
+    const policy = new iam.ManagedPolicy(this, 'Resource', {
+      // A CF update of the description property of a managed policy requires
+      // a replacement. Use the table name in the description to force a managed
+      // policy replacement when the table name changes. This way we preserve permissions
+      // to delete old replicas in case of a table replacement.
+      description: `DynamoDB replication managed policy for table ${sourceTable.tableName}`,
```

Comment:
This is an awesome trick! I'll have to remember this one to use in other Custom Resources.

I'm just a little worried about the delete order - since this depends on the Table, won't it be deleted first, before the replica?

Reply:
This is fine because it depends on the role used by `onEventHandler` and `isCompleteHandler`, see also https://github.com/aws/aws-cdk/pull/8224 for more context on this `SourceTableAttachedPolicy`

See also my test https://github.com/aws/aws-cdk/pull/13300#issuecomment-786802544.